File: app/src/main/java/com/organizer/layouts/todo/FolderLayout.java
package com.organizer.layouts.todo;

import android.content.Context;
import android.graphics.drawable.PaintDrawable;
import com.organizer.MainActivity;
import com.organizer.Task;

public class FolderLayout extends TaskLayout
{
    private Task[] tasks;
    
    public FolderLayout(Context context, ToDoLayout parent, String title)
    {
        super(context, parent, title);
        ((PaintDrawable) background.getBackground()).setCornerRadius(MainActivity.getDisplayMetricsController().dpToPx(4));
        setOnClickListener(view -> open());
    }
    
    public FolderLayout(Context context, ToDoLayout parent, FolderLayout folder)
    {
        super(context, parent, folder.title.getText().toString());
        ((PaintDrawable) background.getBackground()).setCornerRadius(MainActivity.getDisplayMetricsController().dpToPx(4));
        setOnClickListener(view -> close());
    }
    
    private void open()
    {
        MainActivity.getInstance().getLayout().getToDoFolderLayout().show(this);
    }
    
    private void close()
    {
        MainActivity.getInstance().getLayout().getToDoFolderLayout().hide();
    }
}

File: app/src/main/java/com/organizer/layouts/todo/FolderLayout.kt
package com.organizer.layouts.todo

import android.annotation.SuppressLint
import android.graphics.drawable.PaintDrawable
import com.organizer.DisplayMetricsController
import com.organizer.MainActivity.Companion.inject
import com.organizer.layouts.MainLayout

@SuppressLint("ViewConstructor")
class FolderLayout : TaskLayout {
    private val displayMetricsController: DisplayMetricsController by inject()
    private val mainLayout: MainLayout by inject()

    constructor(
        parentLayout: ToDoLayout?,
        id: Int,
        title: String
    ) : super(parentLayout!!, id, title)

    constructor(folder: FolderLayout) : super(
        folder.parentLayout,
        folder.identifier,
        folder.title.text.toString()
    )

    init {
        (background.background as PaintDrawable).setCornerRadius(
            displayMetricsController.dpToPx(
                4f
            ).toFloat()
        )
        setOnClickListener { mainLayout.toDoFolderLayout.hide() }
    }
}
